crates/ruff/src/rules/pydocstyle/rules/capitalized.rs

```diff
@@ -59,6 +59,15 @@ pub fn capitalized(checker: &mut Checker, docstring: &Docstring) {
     if first_char.is_uppercase() {
         return;
     };
+    if first_char
+        .to_uppercase()
+        .next()
```

Comment:
Do we know why `to_uppercase` returns an iterator? When would it have 0, or more than 1 element?

Reply:
Because Unicode.

The problem is that there's absolutely no guarantee that converting the case of a single code point will result in a single code point. For example, ["ß" uppercases to "SS"](http://unicode.org/faq/casemap_charprop.html#11).

We avoid checking further if it's not an ASCII alphabet whose code is just a few lines above this, but I think you already updated the code :)